### PR TITLE
feat(exception): add handler for system register access exceptions

### PR DESCRIPTION
**Pull request by luodeb**
- Implement handle_system_register function to process system register access exceptions
- Add new exit reasons for system register read and write operations
- Update exception handling logic to include trapped MSR/MRS instructions
- Add utility functions for extracting system register access information from ISS
- Introduce Aarch64EmuRegs module for system register emulation

### CODE REVIEW

File: src/system_registers.rs
Comment (aarkegz) [inline src/system_registers.rs:1]: This file can be temporarily excluded from this PR.

File: src/exception_utils.rs [hunk 1]
```diff
@@ -171,6 +171,27 @@ pub fn exception_iss() -> usize {
...
+pub fn exception_sysreg_gpr(iss: u64) -> u64 {
+    const ESR_ISS_SYSREG_REG_OFF: u64 = 5;
+    const ESR_ISS_SYSREG_REG_LEN: u64 = 5;
+    const ESR_ISS_SYSREG_REG_MASK: u64 = (1 << ESR_ISS_SYSREG_REG_LEN) - 1;
+    (iss >> ESR_ISS_SYSREG_REG_OFF) & ESR_ISS_SYSREG_REG_MASK
+}
+
+#[inline(always)]
+pub const fn exception_sysreg_addr(iss: usize) -> usize {
+    // (Op0[21..20] + Op2[19..17] + Op1[16..14] + CRn[13..10]) + CRm[4..1]
```
Comment (aarkegz) [inline src/exception_utils.rs:190]: Please update the comment here.

Comment (aarkegz) [reply src/exception_utils.rs:190]: As in [here](https://github.com/arceos-hypervisor/aarch64_sysreg/blob/main/src/system_reg_type.rs).

**Review by aarkegz — CHANGES_REQUESTED** [review]

File: src/system_registers.rs
Comment (luodeb) [reply src/system_registers.rs:1]: ok

**Pull request merged**